src/domchange.js

```diff
@@ -82,18 +83,40 @@ export class DOMChange {
     this.view.inDOMChange = null
   }
 
-  compositionEnd() {
+  compositionEnd(event) {
     if (this.composing) {
       this.composing = false
+      if (event) this.compositionEndedAt = event.timeStamp
       this.timeout = setTimeout(() => this.finish(), 50)
     }
   }
 
+  ignoreKeyDownOnCompositionEnd(event) {
+    // See https://www.stum.de/2016/06/24/handling-ime-events-in-javascript/.
+    // On Japanese input method editors (IMEs), the Enter key is used to confirm character
+    // selection. On Safari, when Enter is pressed, compositionend and keydown events are
+    // emitted. The keydown event triggers newline insertion, which we don't want.
+    // This method returns true if the keydown event should be ignored.
+    // We only ignore it once, as pressing Enter a second time *should* insert a newline.
+    // Furthermore, the keydown event timestamp must be close to the compositionEndedAt timestamp.
+    // This guards against the case where compositionend is triggered without the keyboard
+    // (e.g. character confirmation may be done with the mouse), and keydown is triggered
+    // afterwards- we wouldn't want to ignore the keydown event in this case.
+    if (this.compositionEndedAt !== null) {
+      if (Math.abs(event.timeStamp - this.compositionEndedAt) < 200) {
```

Comment:
This could be a single `if` with `&&` instead of a nested one. And does it make sense to only do this check on Safari, as further protection against false positives?

Reply:
Thanks! I've combined the nested `if`s and added Safari check. I've also increased the delta time from 200 to 500, as upon re-testing my changes I got a delta of 400 one time.